### ISSUE TITLE
Downsampling the object point cloud uniformly. Code not found. 

### ISSUE THREAD

**Issue by kavnat** [charlesq34/pointnet#149]
Thanks for the work on this, authors! I have been trying to create my own dataset using point clouds to train and evaluate its performance. I want to downsample object point clouds down to 256 points. However, I could not find any code in the repo that does that. The authors mentioned sampling like this: 

"We uniformly sample 1024 points on mesh faces according to face area and normalize them into a unit sphere. During training we augment the point cloud on-the-fly by randomly rotating the object along the up-axis and jitter the position of each points by a Gaussian noise with zero mean and 0.02 standard deviation."

Can anyone please help me understand what exactly does it mean and if there is any code available out there that can help me sample point clouds like this? Currently, I am using numpy for downsampling and the results are not good at all.

**Comment by rejivipin** [comment]
@seanhuang5104  As explained above I understand the scales should be of the same range. But why such a sampling is needed for classification procedure? Why cant the same process that is used in semantic segmentation to generate the hdf5 datasets

**Comment by seanhuang5104** [comment]
@rejivipin I assume you mean why need pcl_mesh_sampling.If you visualize the raw data from modelnet, they only contains the points at corners(for example, a simple box will only has 8 data points) which is not enough data for the model to learn the overall shape. Points outside corners are important since it provide info regarding edges and surfaces that defined the shape as well. pcl_mesh_sampling help you extract point data from model surfaces not only corner of it.

**Comment by rejivipin** [comment]
@seanhuang5104 Exactly and a big thanks, You cleared my doubt. If the data we are considering is vegetation, then mesh sampling might not be a solution. Right? In such a case how will we generate the hdf5 files?

**Comment by seanhuang5104** [comment]
Your data is vegetation? Not sure I get that, mind explained a bit more since maybe its domain specific.

**Comment by rejivipin** [comment]
yes... I am working on crops. Crops don't have a regular geometry. My query is, for such a dataset, how will be the input data file generated?
